### PR TITLE
Simple Animal Bots - Beepsky strikes back.

### PR DESCRIPTION
**Pull request by AnturK**
Alright let's try this again.

### CODE REVIEW

File: code/game/machinery/portable_turret.dm [hunk 1]
```diff
@@ -448,7 +448,9 @@
 
 	if(check_anomalies)	//if it's set to check for xenos/simpleanimals
 		for(var/mob/living/simple_animal/SA in turretview)
-			if(!SA.stat && (!SA.has_unlimited_silicon_privilege || !(faction in SA.faction)) ) //don't target dead animals or NT maint drones.
+			if(isbot(SA))
+				continue
+			if(!SA.stat && (!SA.has_unlimited_silicon_privilege || !(faction in SA.faction))) //don't target dead animals or NT maint drones.
```
Comment (phil235) [inline code/game/machinery/portable_turret.dm]: Is there a reason not to give has_unlimited_silicon_privilege to bots to simplify this?

Comment (AnturK) [inline code/game/machinery/portable_turret.dm]: Otherwise beepsky/secbots/mulebots would be able to open all doors on their path.

File: code/modules/mob/living/carbon/life.dm [hunk 2]
```diff
@@ -69,6 +69,8 @@
 
 				breath = loc.remove_air(breath_moles)
 
+				breath = loc.remove_air(breath_moles)
```
Comment (phil235) [inline code/modules/mob/living/carbon/life.dm]: why is this here?

Comment (AnturK) [inline code/modules/mob/living/carbon/life.dm]: Hum, something went wrong during rebase. Will remove in a moment.

**Comment by Dorsisdwarf** [comment]
I just cant help but feel that in a month's time there's going to a be a PR which lets players be these all-access infinite-power stun-on-bump unsubvertable mobs

**Comment by phil235** [comment]
Does this allow sentience potions to work on bot? Can gold slime spawn bots?

**Comment by AnturK** [comment]
No, it specifically add sentience types for that. I will add it to other mobs that shouldn't be potioned later.

**Comment by tkdrg** [comment]
This is a great improvement because it reduces the special case handling for bots and brings them in line with mostly everything else in our code. They should never have been objs in the first place. We should not reject an improvement just because somebody might do something stupid in the future. How about we stop those future people from being stupid instead?

**Comment by KorPhaeron** [comment]
So what you're saying is we need beepsky as a roundstart job

**Comment by Dorsisdwarf** [comment]
Oh, by the way, does this mean projectiles no longer pass over bots?

**Comment by AnturK** [comment]
Yes for secbots and mulebots, still no for other types.

File: code/modules/mob/living/simple_animal/bot/bot.dm [hunk 3]
```diff
@@ -1,17 +1,53 @@
...
-	use_power = 0
-	var/obj/item/weapon/card/id/botcard			// the ID card that the bot "holds"
+	stop_automated_movement = 1
+	wander = 0
+	healable = 0
+	damage_coeff = list(BRUTE = 1, BURN = 1, TOX = 0, CLONE = 0, STAMINA = 0, OXY = 0)
+	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	minbodytemp = 0
+	has_unlimited_silicon_privilege = 1
+	sentience_type = SENTIENCE_ARTIFICIAL
```
Comment (phil235) [inline code/modules/mob/living/simple_animal/bot/bot.dm]: Are you sure it doesn't need some pass_flags and status_flag (for example so it cant be stunned) ?
Also, mob_size should probably be MOB_SIZE_SMALL except for mulebot and ED209.

Comment (AnturK) [inline code/modules/mob/living/simple_animal/bot/bot.dm]: Will change the mob size, the non sec ones could probably use PASSMOB yea.

File: code/modules/mob/living/simple_animal/bot/bot.dm [hunk 4]
```diff
@@ -1,17 +1,53 @@
...
+/mob/living/simple_animal/bot
 	icon = 'icons/obj/aibots.dmi'
 	layer = MOB_LAYER
 	luminosity = 3
-	use_power = 0
-	var/obj/item/weapon/card/id/botcard			// the ID card that the bot "holds"
+	stop_automated_movement = 1
+	wander = 0
+	healable = 0
+	damage_coeff = list(BRUTE = 1, BURN = 1, TOX = 0, CLONE = 0, STAMINA = 0, OXY = 0)
```
Comment (Anonus) [inline code/modules/mob/living/simple_animal/bot/bot.dm]: This isn't set right, 1 is for ignored damage types, 0 for allowed types.
Which is stupid, but hey.

**Comment by Dorsisdwarf** [comment]
Secbots should still have the projectiles pass-over without a direct click like they're supposed to.
That's the only thing which lets beepsky have a chance at anyone other than revs

File: code/modules/mob/living/simple_animal/bot/bot.dm (hunk 4 above)
Comment (AnturK) [inline code/modules/mob/living/simple_animal/bot/bot.dm]: This PR changes that to coefficients of damage.

Comment (Anonus) [inline code/modules/mob/living/simple_animal/bot/bot.dm]: Oh okay that's wise and how it should have been anyway :+1:

File: code/modules/mob/living/simple_animal/revenant/revenant_abilities.dm [hunk 5]
```diff
@@ -139,26 +139,26 @@
...
-						corpseholder.open()
-				for(var/obj/machinery/dna_scannernew/dna in T.contents)
-					dna.open_machine()
+					for(var/obj/structure/closet/closet in T.contents)
+						closet.open()
+					for(var/obj/structure/bodycontainer/corpseholder in T.contents)
+						if(corpseholder.connected.loc == corpseholder)
+							corpseholder.open()
+					for(var/obj/machinery/dna_scannernew/dna in T.contents)
+						dna.open_machine()
```
Comment (Anonus) [inline code/modules/mob/living/simple_animal/revenant/revenant_abilities.dm]: Why indent any of these(lines 142 to 150 and 155 to 161)? Indenting them causes problems and makes the ability work improperly.

Comment (Anonus) [inline code/modules/mob/living/simple_animal/revenant/revenant_abilities.dm]: also this reminds me I need to clean these checks up at some point ugh

Comment (AnturK) [inline code/modules/mob/living/simple_animal/revenant/revenant_abilities.dm]: Something went wrong during rebasing here i had to reindent stuff. I'll probably reset this file in the next commit.

**Comment by KorPhaeron** [comment]
>merge conflict

It begins

**Comment by KorPhaeron** [comment]
@tgstation/commit-access

Can you please make reviewing/merging this a priority so it doesn't die to conflicts again?

**Comment by WJohn** [comment]
The conflicts are real.

You're secretly a masochist inside, aren't you?

This PR needs some serious merge priority, given how easily it can conflict with ANYTHING else.

**Comment by KorPhaeron** [comment]
@WJohn 

Go rogue and merge it as soon as he updates it.

Also attach a priority tag to it or something

**Comment by phil235** [comment]
@AnturK you need to add the flag that prevents people from grabbing the mob, we don't want throwable ED209s. And we don't want to be able to choke bots ("Kyra Gray starts to tighten her grip on the ED-209 Security Robot's neck!").

**Comment by AnturK** [comment]
Ugh. For some reason the damage_coeff changes disappeared. I'll fix the grab/pass stuff tommorow.

**Comment by duncathan** [comment]
>`qdel(null)` is runtiming

That's odd

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 6]
```diff
@@ -67,6 +67,8 @@
 
 	var/mob/living/simple_animal/hostile/spawner/nest
 
+	var/sentience_type = SENTIENCE_ORGANIC // Sentience type, for slime potions
```
Comment (phil235) [inline code/modules/mob/living/simple_animal/simple_animal.dm]: is it really necessary to have that var? Right now it's only used for bots so we could just use has_unlimited silicon privilege instead. Unless we plan to restrict sentience potions for other animals in the future.

**Comment by Steelpoint** [comment]
You either need to massively buff the Secbots health or make projectiles pass over them intentionally.

The only thing keeping them alive is the fact projectiles pass over them, otherwise they have very low health and will become a non-issue to anyone with a gun who just blindly shoots down the corridor.

File: code/modules/mob/living/simple_animal/simple_animal.dm (hunk 6 above)
Comment (AnturK) [inline code/modules/mob/living/simple_animal/simple_animal.dm]: I wanted to expand this to human mobs in some other pr. This is mostly for future use, for example away boss monsters and such.

**Comment by phil235** [comment]
I think beepsky is probably better with density = 0, otherwise humans not on HELP intent can't push it (because of the lack of CANSTUN status flag). A better solution would probably be to have use a different flag for non grabbable mobs, but it's work for another PR.

**Comment by AnturK** [comment]
I switched beepsky to density = 0 in last commit. And i was considering splitting the flag but i'd rather see this finished first.
